src/PostgREST/RangeQuery.hs

```diff
@@ -38,12 +41,13 @@ rangeParse range = do
       rangeIntersection lower upper
     Nothing -> rangeGeq 0
 
-rangeRequested :: RequestHeaders -> NonnegRange
-rangeRequested = rangeParse . fromMaybe "" . lookup hRange
+rangeRequested :: RequestHeaders -> Maybe NonnegRange
```

Comment:
Actually won't `limitF` use `limit ALL` if `rangeLimit` returns Nothing? You may be able to unconditionally insert the range in `setTopLevelRange`.